pkg/app/piped/cmd/piped/piped.go

```diff
@@ -630,6 +653,25 @@ func (p *piped) insertLoginUserToPasswd(ctx context.Context) error {
 	return nil
 }
 
+func (p *piped) getConfigDataFromSecretManager(ctx context.Context) ([]byte, error) {
+	client, err := secretmanager.NewClient(ctx)
```

Comment:
I don't know exactly but can we access SecretManage with only the resource ID? Doesn't it need any kind of credentials like [WithCredentials](https://pkg.go.dev/google.golang.org/api/option#WithCredentials) or [WithAPIKey](https://pkg.go.dev/google.golang.org/api/option#WithAPIKey)?

Reply:
The necessary roles can be assigned to that VM directly (or via the service account used to start up it). So we just need the name.

Reply:
That makes sense 👍